sdks/java/core/src/main/java/org/apache/beam/sdk/util/IOChannelFactory.java

```diff
@@ -103,4 +103,9 @@
 
   /** Converts the given string to a {@link Path}. */
   Path toPath(String path);
+
+  /**
+   * Delete files with the given {@code uris} if they exist.
+   */
+  void deleteIfExists(Collection<String> uris) throws IOException;
```

Comment:
We are doing bulk deletes here. deleteAndThrowIfMissing() is hard to use as a primitive operation. That would requires returning 1. missing file names and 2. files deleted successfully.

This is why I picked deleteIfExists() as the primitive.